### PR TITLE
Added new extension to get ST node from reference position

### PR DESCRIPTION
**Pull request by axewilledge**
Fixes [#39118](https://github.com/ballerina-platform/ballerina-lang/issues/39118)

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/extensions/document/SyntaxTreeNodeByPositionTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
```
Comment (NipunaRanasinghe) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/extensions/document/SyntaxTreeNodeByPositionTest.java:2]:
```suggestion
 *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/TextDocumentRequest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (NipunaRanasinghe) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/TextDocumentRequest.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaDocumentService.java [hunk 3]
```diff
@@ -519,4 +525,63 @@ public CompletableFuture<BallerinaSyntaxTreeResponse> resolveMissingDependencies
...
+            } catch (Throwable e) {
+                String msg = "Operation 'ballerinaDocument/syntaxTreeNodeByPosition' failed!";
+                this.clientLogger.logError(DocumentContext.DC_NODE_DEFINITION_BY_POSITION,
+                        msg, e, params.getTextDocument(),
+                        params.getPosition());
+            }
+            return reply;
+        });
+
+
```
Comment (NipunaRanasinghe) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaDocumentService.java:585]: Additional blank lines

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by IMS94 — APPROVED** [review]
Please create an issue and track using workspaceManagerProxy.get(fileUri) method instead of using the get() method.

**Comment by axewilledge** [comment]
Refactor issue created: https://github.com/ballerina-platform/ballerina-lang/issues/39220

**Review by NipunaMadhushan — APPROVED** [review]

**Pull request merged**